Add dark theme to web3 modal

amochuko opened the pull request:
<!--- Please describe what this PR is about here. -->


## Related GitHub Issue

<!--- Please link to the GitHub issue here. -->
closes #https://github.com/nation3/citizen-app/issues/310

## Screenshots (if appropriate):

<!--- If your pull request changes the UI, please include before/after screenshots. -->

_Before_ code change:
> <img width="447" alt="Screenshot 2024-08-27 at 09 39 14" src="https://github.com/user-attachments/assets/12c31d5a-b262-498a-ba27-95c3fcfcb115">

_After_ code change:
><img width="440" alt="Screenshot 2024-08-27 at 12 16 41" src="https://github.com/user-attachments/assets/b41dee9c-f632-45a2-9fe1-0298d3600a6b">

## How Has This Change Been Tested?

<!--- Please describe in detail how you tested your changes. -->


<!-- Tick the checklist for the tests you completed: -->
- [x] All [status checks](https://github.com/nation3/citizen-app/blob/main/.github/workflows/ui_mainnet.yml#L35) pass (build, lint, e2e, test)
- [x] Works on Sepolia preview deployment
- [x] Works on Mainnet preview deployment

## Are Any Admin Tasks Required?

<!--- Please include any related admin tasks, like adding/changing environment variables in Vercel. -->
- [x] No admin tasks

amochuko committed:
update with new install

amochuko committed:
add dark theme to web3 modal

amochuko committed:
remove border

aahna-ashina commented on ui/components/Layout.tsx:
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
 
-              <ul className="menu bg-base-100 p-2 -m-2 rounded-box">
-                <li key="address">
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box space-y-2 rounded-box dark:bg-slate-800">
```
If the light theme does not have vertical spacing, do not add vertical spacing to the dark theme.
```suggestion
              <ul className="menu bg-base-100 p-2 -m-2 rounded-box rounded-box dark:bg-slate-800">
```

aahna-ashina commented on ui/components/Layout.tsx:
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box space-y-2 rounded-box dark:bg-slate-800">
+                <li
+                  key="address"
+                  className="dark:border-[1px] dark:border-slate-600 my-1 dark:bg-slate-700
```
If the light theme does not have a border, do not add a border to the dark theme.
```suggestion
                  className="my-1 dark:bg-slate-700
```

amochuko committed:
Update ui/components/Layout.tsx

accept suggestion to remove vertical spacing (not available in light theme)

Co-authored-by: Aahna Ashina <95955389+aahna-ashina@users.noreply.github.com>

amochuko committed:
Update ui/components/Layout.tsx

Accept suggestion to remove border from ul tag when wallet is connected

Co-authored-by: Aahna Ashina <95955389+aahna-ashina@users.noreply.github.com>

amochuko committed:
remove duplicate class

amochuko committed:
remove border

aahna-ashina commented on ui/components/Layout.tsx:
```diff
@@ -297,12 +311,14 @@ export default function Layout({ children }: any) {
                     <button
                       disabled={!connector.ready}
                       onClick={() => connect({ connector })}
+                      className="dark:border-[1px] dark:border-slate-600 my-1 dark:bg-slate-700
```
```suggestion
                      className="dark:bg-slate-700
```

aahna-ashina commented on ui/components/Layout.tsx:
```diff
@@ -271,7 +281,11 @@ export default function Layout({ children }: any) {
-                <li key="logout">
+                <li
+                  key="logout"
+                  className="my-1 dark:bg-slate-700
+                      dark:hover:bg-slate-600 rounded-md"
```
```suggestion
                  className="dark:bg-slate-700
                      dark:hover:bg-slate-600"
```

aahna-ashina commented on ui/components/Layout.tsx:
```diff
@@ -254,8 +260,12 @@ export default function Layout({ children }: any) {
+              <ul className="menu bg-base-100 p-2 -m-2 rounded-box dark:bg-slate-800">
+                <li
+                  key="address"
+                  className="my-1 dark:bg-slate-700
+                      dark:hover:bg-slate-600 rounded-md"
```
```suggestion
                  className="dark:bg-slate-700
                      dark:hover:bg-slate-600"
```

aahna-ashina commented on ui/yarn.lock:
@amochuko Please resolve the merge conflicts for this file.

aahna-ashina reviewed (changes requested):
Merge conflicts in `yarn.lock`.

amochuko committed:
Update ui/components/Layout.tsx

Co-authored-by: Aahna Ashina <95955389+aahna-ashina@users.noreply.github.com>

amochuko committed:
Update ui/components/Layout.tsx

Co-authored-by: Aahna Ashina <95955389+aahna-ashina@users.noreply.github.com>

amochuko committed:
Update ui/components/Layout.tsx

Co-authored-by: Aahna Ashina <95955389+aahna-ashina@users.noreply.github.com>

amochuko committed:
Merge branch 'main' of https://github.com/amochuko/citizen-app into add-dark-theme-to-modal to resolve yarn.lock conflicts

amochuko reviewed (commented):
Kindly review the yarn.lock conflict resolve

amochuko committed:
Revert "Merge branch 'main' of https://github.com/amochuko/citizen-app into add-dark-theme-to-modal to resolve yarn.lock conflicts"

This reverts commit a8f3d3f32d0f4f5e866d6780061e6876ad8e7b23, reversing
changes made to edadb3b2494f278dd029ce59272fa36c9e682743.

dependabot[bot] committed:
chore(deps): bump jest and @types/jest in /ui

Bumps [jest](https://github.com/jestjs/jest/tree/HEAD/packages/jest) and [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest). These dependencies needed to be updated together.

Updates `jest` from 28.1.3 to 29.7.0
- [Release notes](https://github.com/jestjs/jest/releases)
- [Changelog](https://github.com/jestjs/jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jestjs/jest/commits/v29.7.0/packages/jest)

Updates `@types/jest` from 28.1.8 to 29.5.12
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: "@types/jest"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

aahna-ashina committed:
Merge pull request #320 from nation3/dependabot/npm_and_yarn/ui/multi-7c1d38dd65

chore(deps): bump jest and @types/jest in /ui

dependabot[bot] committed:
chore(deps): bump next from 14.2.7 to 14.2.13 in /ui

Bumps [next](https://github.com/vercel/next.js) from 14.2.7 to 14.2.13.
- [Release notes](https://github.com/vercel/next.js/releases)
- [Changelog](https://github.com/vercel/next.js/blob/canary/release.js)
- [Commits](https://github.com/vercel/next.js/compare/v14.2.7...v14.2.13)

---
updated-dependencies:
- dependency-name: next
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

aahna-ashina committed:
Merge pull request #326 from nation3/dependabot/npm_and_yarn/ui/next-14.2.13

chore(deps): bump next from 14.2.7 to 14.2.13 in /ui

aahna-ashina commented on ui/package.json:
```diff
@@ -29,7 +29,7 @@
     "eslint-config-prettier": "^9.1.0",
     "ethers": "^5.7.1",
     "jest": "^28.1.3",
-    "next": "14",
```
@amochuko Is it necessary to downgrade Next.js?

amochuko replied:
No it's not. There isn't any need doing so.

aahna-ashina commented:
> Kindly review the yarn.lock conflict resolve 

@amochuko Your branch still has merge conflicts from what I can see.

![](https://github.com/user-attachments/assets/8e90ece0-33c1-404c-a816-b472e694dee9)

amochuko committed:
Merge branch 'add-dark-theme-to-modal' into main-two to resolve conflict of latest pull

amochuko committed:
fix: add mb-0.5 for visuals
